src/DynamicData/Cache/Internal/DisposeMany.cs

```diff
@@ -49,7 +49,7 @@ public DisposeMany(IObservable<IChangeSet<TObject, TKey>> source, Action<TObject
 
     private void RegisterForRemoval(IChangeSet<TObject, TKey> changes, Cache<TObject, TKey> cache)
     {
-        changes.ForEach(
+        changes.ToConcreteType().ForEach(
```

Comment:
I think this should be changed to the traditional foreach() loop instead of the ForEach extension.

I wrote the ForEach extension, but it was misguided as it also allocates.  However, looking at the usages:

![image](https://github.com/reactivemarbles/DynamicData/assets/9879866/8cd50a08-62cf-4b5f-90de-fd05844bf400)

That should be changed in a distinct PR

Reply:
Yeah, I can do a distinct PR for that.  I'd like to understand more about why it allocates because that is news to me.  As a fan of `ForEach`, maybe it can be updated to avoid the allocation instead of needing to be removed.